datacache/cache.py

```diff
@@ -16,10 +16,11 @@ def __init__(self, subdir="datacache"):
         self.local_paths = {}
 
     def delete_key(self, url):
-        path = self.local_paths(url)
-        del self.local_paths[url]
-        if exists(path):
-            remove(path)
+        if url in self.local_paths:
+            path = self.local_paths[url]
+            del self.local_paths[url]
+            if exists(path):
+                remove(path)
```

Comment:
It's been a while since I looked at this code so I'm not sure what my intended semantics were. However, wouldn't it make more sense to make removing the `url` from `local_paths` and deleting `path` separate?

```
    def delete_key(self, url,  filename=None, decompress=False):
        if url in self.local_paths:
           del self.local_paths[url]
        path = self.local_path(url, filename, decompress)         
        if exists(path):
            remove(path)
```